src/assistant/documentService.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import puppeteer from 'puppeteer';
```

Comment:
Check out the content-scraper's use of Plugins: https://github.com/adobe/spacecat-content-scraper/blob/main/src/handlers/abstract-handler.js#L28

Anything we can/should reuse here?

Reply:
Incliuding plugins would require using `puppeteer-extra` instead of just `puppeteer`. So far I haven't come across a case where these plugins would be needed but I am sure there are sites out there that may be more restrictive to warrant their use.

Reply:
Could https://spacecat.experiencecloud.live/api/v1/scrape be used instead?